PedestriamTrafficLighterShowModule.cs

```diff
@@ -4,21 +4,62 @@
 using System.Text;
 using System.Threading.Tasks;
 using System.Xml.Linq;
+using static Traffic_lighters.CrossRoadController;
 
 namespace Traffic_lighters
 {
     internal class PedestrianTrafficLighterEventArgs
     {
-        internal bool RedLamp { get; set; }
-        internal bool GreenLamp { get; set; }
-        internal PedestrianTrafficLighterEventArgs(bool redLamp, bool greenLamp)
+        internal CrossRoadController.StatesCondition? State { get; set; }
+        internal CrossRoadController.TrafficLightsNames Name { get; set; }
+        internal int? Time { get; set; }
+        internal PedestrianTrafficLighterEventArgs(CrossRoadController.StatesCondition? state, CrossRoadController.TrafficLightsNames name, int? time)
         {
-            RedLamp = redLamp;
-            GreenLamp = greenLamp;
+            State = state;
+            Name = name;
+            Time = time;
         }
-        internal static void ShowPedestrianTrafficLight(Pedestrian_Traffic_Lighters pedestrianTrafficLighter, PedestrianTrafficLighterEventArgs e)
+        internal bool RedLamp { get; set; }
+        internal bool GreenLamp { get; set; }
+        internal static int blinkDuratation = 500;
+        internal static void ShowPedestrianTrafficLight(PedestrianTrafficLighterEventArgs e)
         {
-            Console.WriteLine($"{pedestrianTrafficLighter.Name}");
+            async void BlinkGreen()
+            {
+                e.RedLamp = false;                
+                for (int i = 0; i < e.Time; i += blinkDuratation)
+                {
+                    if (e.GreenLamp == false)
+                    {
+                        e.GreenLamp = true;
+                        await Task.Delay(blinkDuratation);
+                    }
+                    else
+                    {
+                        e.GreenLamp = false;
+                        await Task.Delay(blinkDuratation);
+                    }
+                }
+            }
+            switch (e.State)
+            {
+                case StatesCondition.RED:
+                    e.RedLamp = true;
+                    e.GreenLamp = false;
+                    break;                
+                case StatesCondition.GREEN:
+                    e.RedLamp = false;
+                    e.GreenLamp = true;
+                    break;
+                case StatesCondition.BLINKGREEN:
+                    BlinkGreen();
+                    break;
+                case StatesCondition.OFFLIGHT:
+                    e.RedLamp = false;
+                    e.GreenLamp = false;
+                    break;
+            }
```

Comment:
Above you set a state of the traffic lights.
And below you show that state in the console.
Reread my comment from the previous pull request:
> TrafficLightі should not know how to present itself on a UI. In WinForms application thare will be no Console. Let TrafficLightPresenter to decide how to present the TrafficLight of the UI.

Reply:
Again you mistake is that you combined two entities "TrafficLight" and "Presenter" in one class. And in addition to that you gave the name to the class that is related neither to "TrafficLight" nor "Presenter": PedestrianTrafficLighterEventArgs.

To understand how to implement the system describe it in human words:

We have _**trafficlights**_ on _**crossroad**_ that traffic lights are working together, so there should be some _**controller**_ that controls the situation (or state) on that crossroad. The _**state of the crossroad**_ defines _**state of each traffic light**_". _**Lamps**_ of a specific traffic light shows a state of that traffic light.

I highlited nouns in the text and those nowns are potential names of the classes that you need in the system. Ofcouse some of them need better names, for example instead of "Lamps" you can use "TrafficLightPresenter" or "TrafficLightView" but the meaning of the term should be the same.